packages/running-extension/src/kernels.tsx

```diff
@@ -49,6 +58,59 @@ export async function addKernelRunningSessionManager(
   // Wait until the relevant services are ready.
   await Promise.all([kernels.ready, kernelspecs.ready, sessions.ready]);
 
+  async function getUnusedKernels() {
+    // Identifies unused kernels
+    const runningKernels = await KernelAPI.listRunning();
+    return runningKernels.filter(
+      kernel => kernel.connections !== undefined && kernel.connections < 1
+    );
```

Comment:
Is this information possibly available from `kernels` service already?

Reply:
It is, though when I changed it to Array.from(kernels.running()), the code stopped working
every kernel would always get shutdown, regardless of connection count

I'm not sure why, If I print both I don't see a difference?

I reverted it back to await KernelAPI.listRunning() which works correctly

await KernelAPI.listRunning()
```
[
  {
    "id": "5349a541-b573-4c75-aaf5-efd484103e5f",
    "name": "python3",
    "last_activity": "2024-06-04T01:37:31.710898Z",
    "execution_state": "idle",
    "connections": 0
  },
  {
    "id": "1686c8f7-dc10-4122-938c-0ae73c5cc3f6",
    "name": "python3",
    "last_activity": "2024-06-04T01:37:39.079313Z",
    "execution_state": "idle",
    "connections": 1
  }
]
```

Array.from(kernels.running())
```
[
  {
    "id": "5349a541-b573-4c75-aaf5-efd484103e5f",
    "name": "python3",
    "last_activity": "2024-06-04T01:37:31.710898Z",
    "execution_state": "idle",
    "connections": 0
  },
  {
    "id": "1686c8f7-dc10-4122-938c-0ae73c5cc3f6",
    "name": "python3",
    "last_activity": "2024-06-04T01:37:39.079313Z",
    "execution_state": "idle",
    "connections": 1
  }
]
```